mlir/lib/Dialect/LLVMIR/IR/LLVMMemorySlot.cpp

```diff
@@ -590,6 +590,30 @@ DeletionKind LLVM::InvariantEndOp::removeBlockingUses(
   return DeletionKind::Delete;
 }
 
+bool LLVM::LaunderInvariantGroupOp::canUsesBeRemoved(
```

Comment:
Can you add a test for this intrinsics that test if mem2reg can drop them. A test similar to this one should work:
https://github.com/llvm/llvm-project/blob/7302c8dbe71b7c03b73a35a21fa4b415fa1f4505/mlir/test/Dialect/LLVMIR/mem2reg.mlir#L406

Reply:
Didn't really take care into this and the implementation was incorrect. Updated.